app/script/view_model/ConversationInputEmojiViewModel.js

```diff
@@ -0,0 +1,175 @@
+        this.rotate_emoji_list(event.keyCode === z.util.KEYCODE.ARROW_UP);
+        this.suppress_key_up = true;
+        break;
+      case z.util.KEYCODE.ENTER: {
+        let input = event.target;
+        let emoji = this.emoji_list.find('>div.selected>span').html();
+        let val = input.value;
+        input.value = val.substr(0, this.emoji_start_pos - 1) + emoji + val.substr(input.selectionStart);
+        input.setSelectionRange(this.emoji_start_pos, this.emoji_start_pos);
+        this.emoji_list.remove();
+        this.emoji_start_pos = -1;
+        $(input).change();
+        break;
+      }
+      default:
+        return false;
+    }
+
+    event.preventDefault();
+    return true;
+  }
+
+  on_input_key_up(data, event) {
+    if (!this.suppress_key_up) {
+      let input = event.target;
+      let text = input.value || '';
+
+      if (text[input.selectionStart - 1] === ':') {
+        this.emoji_start_pos = input.selectionStart;
+        this.update_emoji_list(input);
+      } else if (this.emoji_start_pos !== -1) {
+        if (input.selectionStart < this.emoji_start_pos || text[this.emoji_start_pos - 1] !== ':') {
+          this.emoji_list.remove();
+          this.emoji_start_pos = -1;
+        } else {
+          this.update_emoji_list(input);
+        }
+      }
+    }
+
+    this.suppress_key_up = false;
+    return true;
+  }
+
+  update_emoji_list(input) {
+    if (!input.value) {
+      return;
+    }
+
+    let query = input.value.substr(this.emoji_start_pos, input.selectionStart - this.emoji_start_pos);
+    if (query.length < ConversationInputEmojiViewModel.MIN_QUERY_LENGTH || query[0] === ' ' || !this.emoji_dict) {
+      this.emoji_list.remove();
+    } else {
+      let emoji_matched = this.emoji_dict
+        .filter((emoji) => emoji.indexOf(query) !== -1)
+        .slice(0, ConversationInputEmojiViewModel.EMOJI_LIST_LENGTH)
+        .map((emoji) => {
+          let [code, name] = emoji.split('\t');
+          let parsed_unicode_emoji = String.fromCodePoint.apply(null, code.split(','));
+          return `<div><span>${parsed_unicode_emoji}</span>${name}</div>`;
+        })
+        .join('');
+
+      if (emoji_matched === '') {
+        this.emoji_list.remove();
+      } else {
+        this.emoji_list.html(emoji_matched).appendTo('body').show();
+        this.emoji_list.find('>div:nth(0)').addClass('selected');
+
+        let pos = this.get_cursor_pixel_pos(input);
+        let top = pos.top - this.emoji_list.height();
+
+        this.emoji_list.css('left', pos.left);
+        this.emoji_list.css('top', top);
+      }
+    }
+  }
+
+  rotate_emoji_list(backward) {
+    let previous = this.emoji_list.find('>div.selected');
+    let new_selection = (previous.index() + (backward ? -1 : 1)) % this.emoji_list.find('>div').length;
+    previous.removeClass('selected');
+    this.emoji_list.find(`>div:nth(${new_selection})`).addClass('selected');
+  }
+
+  get_cursor_pixel_pos(input) {
```

Comment:
This is some dark magic... 😉 
but it works well, the emoji list is always positioned correctly from what I've tested so far.